### PR TITLE
Sm/sf command

### PR DESCRIPTION
**Pull request by mshanemc**
switch to use sfCommand, new plugin-cmd-ref.

This is technically a breaking change with log-level disappearing

[@W-12083613@](https://gus.lightning.force.com/a07EE00001DTZCFYA5)

### CODE REVIEW

File: src/commands/cli/latestrc/build.ts [hunk 1]
```diff
@@ -67,27 +69,27 @@ export default class build extends SfdxCommand {
...
     this.exec(`git checkout -b ${nextRCVersion}`);
 
     // bump the version in the pjson to the new latest-rc
-    this.ux.log(`setting the version to ${nextRCVersion}`);
+    this.log(`setting the version to ${nextRCVersion}`);
     repo.package.setNextVersion(nextRCVersion);
     repo.package.packageJson.version = nextRCVersion;
 
-    const only = this.flags.only as string[];
+    const only = flags.only.split(',').map((s) => s.trim());
```
Comment (iowillhoit) [inline src/commands/cli/latestrc/build.ts:89]: This needs to be updated after the changes to sf-plugins-core https://github.com/salesforcecli/sf-plugins-core/pull/155/files#diff-34fb7f2a65843f66e33eb600720236a297d49883585830100e0dac1150768ae5R82

File: src/commands/cli/schemas/collect.ts [hunk 2]
```diff
@@ -22,45 +22,45 @@ const messages = Messages.load('@salesforce/plugin-release-management', 'cli.sch
   'examples',
 ]);
 
-export class SchemaUtils {
```
Comment (iowillhoit) [inline src/commands/cli/schemas/collect.ts:25]: Yay for replacing unnecessary classes 🎉

File: src/commands/channel/promote.ts [hunk 3]
```diff
@@ -10,58 +10,60 @@ import * as os from 'os';
...
     }),
-    target: flags.string({
+    'promote-to-channel': Flags.string({
       char: 't',
       default: Channel.STABLE,
-      description: messages.getMessage('target'),
+      summary: messages.getMessage('target'),
       // options: Object.values(Channel),
       required: true,
+      aliases: ['target'],
```
Comment (iowillhoit) [inline src/commands/channel/promote.ts:43]: Want to add `deprecateAliases: true` for the flags you've renamed?

**Review by iowillhoit — APPROVED** [review]

File: src/package.ts [hunk 4]
```diff
@@ -92,9 +92,9 @@ export class Package extends AsyncOptionalCreatable {
   private nextVersion: string;
   private registry: Registry;
 
-  public constructor(location: string) {
+  public constructor(opts: { location: string }) {
```
[the remark this answers is not in the document]
Comment (mshanemc) [reply src/package.ts:95]: the class being extended expects an object (that changed at some point once TS got stricter) 
TS rightly flagged that we were incorrectly extending it here by not using an object

https://github.com/forcedotcom/kit/blob/886f08fac7beda29f8ecf5aa900cc87b125e81d1/src/creatable.ts#L43
https://github.com/forcedotcom/kit/commit/e2947982366f8318ad395ce682cf8b75d9e69262#diff-c2fce46d25c1ec5f1ee41af2fb21a9863b3ea8dc0dd5084b95aa465e88277174

File: src/commands/channel/promote.ts (hunk 3 above)
Comment (mshanemc) [reply src/commands/channel/promote.ts:43]: nah, I think it'll just create extra noise in the GHA logs.  

We can change our scripts if we ever really want to.

**Pull request merged**